client/state/posts/reducer.js

```diff
@@ -163,40 +170,33 @@ export const queries = ( () => {
 
 	return createReducer( {}, {
 		[ POST_RESTORE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: '__RESTORE_PENDING'
 			}, { patch: true } );
 		},
 		[ POST_RESTORE_FAILURE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: 'trash'
 			}, { patch: true } );
 		},
 		[ POSTS_REQUEST_SUCCESS ]: ( state, { siteId, query, posts, found } ) => {
-			if ( ! state[ siteId ] ) {
-				state = {
-					...state,
-					[ siteId ]: new PostQueryManager()
-				};
-			}
-
-			return applyToManager( state, siteId, 'receive', posts, { query, found } );
+			return applyToManager( state, siteId, 'receive', true, posts, { query, found } );
 		},
 		[ POSTS_RECEIVE ]: ( state, { posts } ) => {
 			return reduce( groupBy( posts, 'site_ID' ), ( memo, sitePosts, siteId ) => {
-				return applyToManager( memo, siteId, 'receive', sitePosts );
+				return applyToManager( memo, siteId, 'receive', true, sitePosts );
 			}, state );
 		},
 		[ POST_SAVE ]: ( state, { siteId, postId, post } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				...post
 			}, { patch: true } );
 		},
-		[ POST_DELETE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'removeItem', postId );
+		[ POST_DELETE_SUCCESS ]: ( state, { siteId, postId } ) => {
+			return applyToManager( state, siteId, 'removeItem', false, postId );
```

Comment:
In `master`, a post disappears from the Trashed list as soon as you click "Delete Permanently" and confirm.  In this branch, the post only disappears when the deletion succeeds.

Also, the popover menu remains active after you click "Delete Permanently":

![image](https://cloud.githubusercontent.com/assets/227022/17116744/bb22a94c-527e-11e6-9007-b0fce5963e8f.png)

The related code in `master` is inconsistent, using `POST_DELETE` sometimes and `POST_DELETE_SUCCESS` other times.  This branch always uses `POST_DELETE_SUCCESS`, which is more consistent by itself, but different from the way trashing and restoring are handled (always optimistically).